### PR TITLE
Feat: Add playfair cipher

### PR DESCRIPTION
**Pull request by JaydityaDhaka**
#### Description of Change

This pull request adds the implementation of the Playfair Cipher algorithm. The Playfair cipher is a digraph substitution cipher, which encrypts pairs of letters from the plaintext using a 5x5 grid. It provides both encryption and decryption functions, handling edge cases such as repeated letters and handling of the 'j' character by replacing it with 'i'.

Key Features:

- Implementation of the encryption and decryption methods.
- A utility to clean input strings by removing non-alphabet characters and converting the text to lowercase.
- Generation of the Playfair 5x5 grid based on a provided key.
- Supports formatted digraphs and insertion of 'x' for repeated characters and padding.

#### Checklist

[x] Added description of change
[x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
[x] Added tests and example, test must pass
[x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
[x] Relevant documentation/comments is changed or added
[x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
[x] Search previous suggestions before making a new one, as yours may be a duplicate.
[x] I acknowledge that all my contributions will be made under the project's license.

Notes: 
This pull request includes a detailed file header using Doxygen-style comments, an example of usage with both encryption and decryption of sample text, and test cases to ensure the correctness of the algorithm.

### CODE REVIEW

**Comment by JaydityaDhaka** [comment]
are there any additional requirments for this PR?

File: ciphers/playfair_cipher.cpp [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+ * The encrypted pairs would then be calculated using the rules above.
+ *
+ * \note This implementation assumes that all text is in lowercase and only handles alphabetic characters ('a'-'z'). The letter 'j' is replaced with 'i'.
+ * 
+ * @author [Jayditya Dhaka]
+ */
+
+
+#include <iostream>
+#include <vector>
```
Comment (realstealthninja) [inline ciphers/playfair_cipher.cpp:37]: document headers please

File: ciphers/playfair_cipher.cpp [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+#include <string>
+#include <cassert>
+
+/** \namespace ciphers
+ * \brief Algorithms for encryption and decryption
```
Comment (realstealthninja) [inline ciphers/playfair_cipher.cpp:42]: use @ tags instead of \ tags

**Review by realstealthninja — CHANGES_REQUESTED** [review]

**Comment by JaydityaDhaka** [comment]
Have put in all the necessary changes

File: ciphers/playfair_cipher.cpp [hunk 3]
```diff
@@ -0,0 +1,241 @@
+/**
+ * @file playfair_cipher.cpp
+ * @brief Implementation of [Playfair cipher](https://en.wikipedia.org/wiki/Playfair_cipher) algorithm.
+ *
+ * @details
+ * The Playfair cipher is a digraph substitution cipher, meaning that it encrypts pairs of letters (digraphs)
```
Comment (realstealthninja) [inline ciphers/playfair_cipher.cpp:6]: lint your file using clang tidy please, Thanks!

File: ciphers/playfair_cipher.cpp [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+    std::cout << "Original text: " << text1;
+    std::cout << " , Encrypted text (key = OLDTAVERN): " << encrypted1;
+    std::cout << " , Decrypted text: " << decrypted1 << std::endl;
+
+    // Test 2
+    std::string text2 = "HELLO";
+    std::string encrypted2 = ciphers::playfair::encrypt(text2, "OLDTAVERN");
+    std::string decrypted2 = ciphers::playfair::decrypt(encrypted2, "OLDTAVERN");
+    std::cout << "Original text: " << text2;
+    std::cout << " , Encrypted text (key = OLDTAVERN): " << encrypted2;
```
Comment (realstealthninja) [inline ciphers/playfair_cipher.cpp:233]: instead of outputing your result check them against the orignal

File: ciphers/playfair_cipher.cpp [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+} // namespace ciphers
+
+void test() {
+    // Test 1
+    std::string text1 = "HEYO";
+    std::string encrypted1 = ciphers::playfair::encrypt(text1, "OLDTAVERN");
+    std::string decrypted1 = ciphers::playfair::decrypt(encrypted1, "OLDTAVERN");
+    std::cout << "Original text: " << text1;
+    std::cout << " , Encrypted text (key = OLDTAVERN): " << encrypted1;
+    std::cout << " , Decrypted text: " << decrypted1 << std::endl;
```
Comment (realstealthninja) [inline ciphers/playfair_cipher.cpp:226]:
```suggestion
   assert(text1 == decrypted1);

```

File: ciphers/playfair_cipher.cpp [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+
+    // Test 2
+    std::string text2 = "HELLO";
+    std::string encrypted2 = ciphers::playfair::encrypt(text2, "OLDTAVERN");
+    std::string decrypted2 = ciphers::playfair::decrypt(encrypted2, "OLDTAVERN");
+    std::cout << "Original text: " << text2;
+    std::cout << " , Encrypted text (key = OLDTAVERN): " << encrypted2;
+    std::cout << " , Decrypted text: " << decrypted2 << std::endl;
```
Comment (realstealthninja) [inline ciphers/playfair_cipher.cpp:234]:
```suggestion
    assert(text2 == decrypted2)

```

**Review by realstealthninja — CHANGES_REQUESTED** [review]

File: ciphers/playfair_cipher.cpp (hunk 3 above)
Comment (realstealthninja) [reply ciphers/playfair_cipher.cpp:6]: file does not seem linted!

**Review by realstealthninja — CHANGES_REQUESTED** [review]
file is still not linted

**Pull request closed without merging**